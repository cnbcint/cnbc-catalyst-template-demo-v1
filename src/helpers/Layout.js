export function getJustifyContent(aligment) {
  switch (aligment) {
    case "left":
      return "flex-start";
    case "right":
      return "flex-end";
    default:
      return "center";
  }
}

export function getAnimation(aligment) {
  switch (aligment) {
    case "left":
      return "fadeInLeft";
    case "right":
      return "fadeInRight";
    default:
      return "fadeInDown";
  }
}

export const breakpoints = {
  mobile: "768px",
  tablet: "1280px",
};

export const margin = {
  base: 1,
  unit: "rem",
};

export function getMargin(multiplyer) {
  return multiplyer * margin.base + margin.unit;
}
