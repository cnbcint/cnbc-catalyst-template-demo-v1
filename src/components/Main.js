import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { getMargin, breakpoints } from "../helpers/Layout";
import { ModalContext } from "../components/Modal";

export const LayoutContext = React.createContext({
  maxWidth: "500px",
  colors: null,
});

const Container = styled.div`
  display: block;
  width: 100%;
  background-color: ${(props) => props.colors.background};
  > * {
    font-size: 1.125rem;
    line-height: 140%;
    color: ${(props) => props.colors.copy};
  }
  hr {
    // border-color: ${(props) => props.primaryColor};
    border-color: ${(props) => props.colors.copy};
    border-style: solid;
    display: inline-block;
    width: 100%;
    // margin-top: ${getMargin(1)};
    margin-bottom: ${getMargin(1)};
    @media (min-width: ${breakpoints.mobile}) {
      // margin-top: ${getMargin(2)};
      margin-bottom: ${getMargin(2)};
    }
  }
  img {
    width: 100%;
    display: block;
    margin-top: ${getMargin(2)};
    margin-bottom: ${getMargin(2)};
  }
  p {
    margin: 1rem auto !important;
  }
  ul {
    list-style: disc !important;
    margin-top: ${getMargin(1)} !important;
    margin-bottom: ${getMargin(1)} !important;
    li {
      list-style: disc !important;
      margin: ${getMargin(1)} !important;
      margin-left: ${getMargin(2)} !important;
    }
  }
`;
export default function Main(props) {
  return (
    <LayoutContext.Provider
      value={{ maxWidth: props.maxWidth, colors: props.colors }}
    >
      <ModalContext.Provider value={{ modals: {} }}>
        <Container
          colors={props.colors}
          primaryColor={props.colors.primary}
          className={props.className}
          style={props.style}
        >
          {props.children}
        </Container>
      </ModalContext.Provider>
    </LayoutContext.Provider>
  );
}
Main.defaultProps = {
  maxWidth: "900px",
  colors: { primary: "black", secondary: "gray", tertiary: "red" },
};
Main.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  maxWidth: PropTypes.string,
  colors: PropTypes.shape({
    primary: PropTypes.string,
    secondary: PropTypes.string,
    tertiary: PropTypes.string,
    background: PropTypes.string,
    copy: PropTypes.string,
  }),
  style: PropTypes.object,
};
