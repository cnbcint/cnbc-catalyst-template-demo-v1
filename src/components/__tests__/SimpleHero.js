import React from "react";
import { render } from "@testing-library/react";
import { css } from "styled-components";
import "jest-styled-components";
import Main from "../Main";
import SimpleHero from "../SimpleHero";
import { breakpoints, getJustifyContent } from "../../helpers/Layout";

const Theme = {
  maxWidth: "900px",
  colors: {
    primary: "orange",
    secondary: "green",
    tertiary: "orange",
    background: "white",
    copy: "black",
    links: "blue",
  },
};
const Component = () => (
  <Main {...Theme}>
    <SimpleHero
      title={() => "A title"}
      subTitle="A sub title"
      copy={() => <p>A tagline</p>}
      companyLogoURL="http://localhost:8080/images/catalyst.svg"
      companyLogoAligment="right"
      copyAligment="center"
      copyMinWidth="640px"
      copyMaxWidth="1280px"
      copyColor="red"
      backgroundImage={{
        desktop: "http://localhost:8080/images/header-desktop.jpg",
        tablet: "http://localhost:8080/images/header-tablet.jpg",
        mobile: "http://localhost:8080/images/header-mobile.jpg",
      }}
      videoId="468693933"
      videoCoverImage="http://localhost:8080/images/header-desktop.jpg"
      addScrollIndicator={true}
      addDivider={true}
      dividerColor="secondary"
      overlayColor="rgba(0,0,0,.5)"
      className="test"
    />
  </Main>
);

beforeAll(() => {
  window.IntersectionObserver = class {
    observe() {
      return null;
    }
    disconnect() {}
  };
});

describe("SimpleHero", () => {
  it("Renders", () => {
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });

  it("Renders the overlay with the correct color", () => {
    const { container } = render(<Component />);

    const target = container.getElementsByClassName("test__overlay")[0];
    expect(target).toHaveStyleRule("background", "rgba(0,0,0,.5)");
  });

  it("Renders the background image", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("test")[0];

    expect(target).toHaveStyleRule(
      "background-image",
      "url(http://localhost:8080/images/header-mobile.jpg)"
    );
  });

  it("Renders the background video", () => {
    const { container } = render(<Component />);
    const target = container.getElementsByClassName("test__video")[0];
    const iframe = target.getElementsByTagName("iframe")[0];
    const src = iframe.getAttribute("src");
    expect(src).toContain("https://player.vimeo.com/video/468693933");
  });

  it("Renders the logo", () => {
    const { container } = render(<Component />);

    const logo = container.getElementsByTagName("img")[0];
    const src = logo.getAttribute("src");
    expect(src).toMatch("http://localhost:8080/images/catalyst.svg");
  });

  it("Logo has alignment style", () => {
    const { container } = render(<Component />);

    const logo = container.getElementsByTagName("img");
    const logoContainer = logo[0].parentElement;
    expect(logoContainer).toHaveStyleRule(
      "justify-content",
      getJustifyContent("right")
    );
  });

  it("Renders the title", () => {
    const { container } = render(<Component />);

    const h1 = container.getElementsByTagName("h1")[0];
    expect(h1.textContent).toMatch("A title");
  });

  it("Renders the sub title", () => {
    const { container } = render(<Component />);

    const h2 = container.getElementsByTagName("h2")[0];
    expect(h2.textContent).toMatch("A sub title");
  });

  it("Renders the copy", () => {
    const { container } = render(<Component />);

    const p = container.getElementsByTagName("p")[0];
    expect(p.textContent).toMatch("A tagline");
  });

  // it("Renders the divider with the correct color", () => {
  //   const { container } = render(<Component />);

  //   const divider = container.getElementsByTagName("hr")[0];
  //   expect(divider).toHaveStyleRule(
  //     "border-color",
  //     `${Theme.colors["secondary"]} !important`
  //   );
  // });

  it("Adds classNames ", () => {
    const { container } = render(<Component />);

    const target = container.getElementsByClassName("test")[0];
    expect(target.classList.contains("test")).toBe(true);
  });

  it("Copy has color style", () => {
    const { container } = render(<Component />);

    const copy = container.getElementsByClassName("test__content");
    expect(copy[0]).toHaveStyleRule("color", "red", {
      modifier: css`> *`,
    });
  });

  it("Copy has alignment style", () => {
    const { container } = render(<Component />);

    const copy = container.getElementsByClassName("test__content");
    expect(copy[0]).toHaveStyleRule("text-align", "center", {
      modifier: css`> *`,
    });
  });

  it("Copy has min-width style", () => {
    const { container } = render(<Component />);

    const copy = container.getElementsByClassName("test__content");
    expect(copy[0]).toHaveStyleRule("min-width", "640px", {
      media: `(min-width:${breakpoints.mobile})`,
    });
  });

  it("Copy has max-width style", () => {
    const { container } = render(<Component />);

    const copy = container.getElementsByClassName("test__content");
    expect(copy[0]).toHaveStyleRule("max-width", "1280px", {
      media: `(min-width:${breakpoints.mobile})`,
    });
  });
});

afterAll(() => {
  window.IntersectionObserver = null;
});
