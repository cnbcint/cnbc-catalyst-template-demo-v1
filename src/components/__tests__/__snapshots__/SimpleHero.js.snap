// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleHero Renders 1`] = `
.c0 {
  display: block;
  width: 100%;
  background-color: white;
}

.c0 > * {
  font-size: 1.125rem;
  line-height: 140%;
  color: black;
}

.c0 hr {
  border-color: black;
  border-style: solid;
  display: inline-block;
  width: 100%;
  margin-bottom: 1rem;
}

.c0 img {
  width: 100%;
  display: block;
  margin-top: 2rem;
  margin-bottom: 2rem;
}

.c0 p {
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul {
  list-style: disc !important;
  margin-top: 1rem !important;
  margin-bottom: 1rem !important;
}

.c0 ul li {
  list-style: disc !important;
  margin: 1rem !important;
  margin-left: 2rem !important;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 2rem;
  position: relative;
  overflow: hidden;
  background-size: cover;
  background-position: center;
  background-image: url(http://localhost:8080/images/header-mobile.jpg);
}

.c8 {
  position: relative;
  padding: 2rem;
  width: calc(100% - 2rem);
  box-sizing: border-box;
}

.c8 > * {
  color: red;
  text-align: center;
}

.c3 {
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  position: absolute;
  background: rgba(0,0,0,.5);
}

.c11 {
  border-style: solid;
  margin: 1rem 0;
  border-color: green !important;
}

.c9 {
  font-size: 1.8rem;
  line-height: 100%;
  font-weight: bold;
  display: inline-block !important;
  width: 100%;
  text-transform: none;
}

.c10 {
  font-size: 1.5rem;
  line-height: 120%;
  font-weight: bold;
}

.c12 {
  font-size: 1rem;
  line-height: 1.4rem;
}

.c6 {
  position: relative;
  max-width: 50%;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1rem;
  box-sizing: border-box;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 0;
  height: 100%;
  background: transparent;
}

.c13 {
  color: white;
  position: absolute;
  bottom: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  font-size: 4rem;
}

.c4 {
  width: 100%;
  height: 100%;
  min-height: 600px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .c0 hr {
    margin-bottom: 2rem;
  }
}

@media (min-width:768px) {
  .c1 {
    padding-bottom: 5rem;
  }
}

@media (min-width:1280px) {
  .c1 {
    padding-bottom: 10rem;
  }
}

@media (min-width:768px) {
  .c1 {
    background-image: url(http://localhost:8080/images/header-tablet.jpg);
  }
}

@media (min-width:1280px) {
  .c1 {
    background-image: url(http://localhost:8080/images/header-desktop.jpg);
  }
}

@media (min-width:768px) {
  .c8 {
    width: 100%;
    min-width: 640px;
    max-width: 1280px;
  }
}

@media (min-width:768px) {
  .c11 {
    margin: 2rem 0;
  }
}

@media (min-width:768px) {
  .c9 {
    font-size: 3.5rem;
  }
}

@media (min-width:768px) {
  .c10 {
    font-size: 2rem;
  }
}

@media (min-width:768px) {
  .c6 {
    width: auto;
  }
}

@media (min-width:768px) {
  .c5 {
    padding: 2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 test"
    >
      <div
        class="c2 test__video"
      >
        <div
          class=""
          style="padding: 0px; position: relative; height: 100%; transform: scale(1);"
        >
          <div
            class=""
            style="background-image: url(http://localhost:8080/images/header-desktop.jpg); position: absolute; width: 100%; height: 100%; top: 0px; background-size: cover;"
          />
          <iframe
            allow="autoplay;"
            allowfullscreen=""
            frameborder="0"
            src="https://player.vimeo.com/video/468693933?background=1&autoplay=1&title=0&byline=0&portrait=0"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%;"
          />
        </div>
      </div>
      <div
        class="c3 test__overlay"
      />
      <div
        class="c4 test__content__container"
      >
        <div
          class="c5 test__logo__container"
        >
          <img
            alt="Logo"
            class="c6 test__logo"
            src="http://localhost:8080/images/catalyst.svg"
          />
        </div>
        <div
          class="c7 test__content__wrapper"
        >
          <div
            class="c8 test__content"
            width="100%"
          >
            <h1
              class="c9 pre-animated"
            >
              A title
            </h1>
            <h2
              class="c10 pre-animated"
            >
              A sub title
            </h2>
            <hr
              class="c11 pre-animated"
            />
            <div
              class="c12 pre-animated"
            >
              <p>
                A tagline
              </p>
            </div>
          </div>
        </div>
        <div
          class="c13 test__more"
        >
          <ion-icon
            name="chevron-down-outline"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
